packages/astro/src/astro-components/interactive/InteractiveUI.astro

```diff
@@ -0,0 +1,42 @@
+---
+interface Props {
+  [key: string]: unknown
+  component: 'sign-in' | 'sign-up' | 'organization-list' | 'organization-profile' | 'organization-switcher' | 'user-button' | 'user-profile'
+}
+
+import { customAlphabet, urlAlphabet } from "nanoid";
+
+const safeId = customAlphabet(urlAlphabet, 10)();
+
+const props = {
+  ...Astro.props,
+}
+---
+
+<div id={`clerk-${props.component}-${safeId}`}></div>
```

Comment:
Do you think this needs to change to something like `data-clerk-id` ?

Reply:
Yeah that's more more sense, I have a question like that [here](https://github.com/clerk/javascript/pull/3656#discussion_r1665978552) in the unstyled components PR. We can go with that 👍🏼

Reply:
Also need to update the mounting function to use that attribute in the query selector

Reply:
that's correct